examples/mercari.py

```diff
@@ -1,9 +1,168 @@
+import argparse
 import os.path as osp
+from typing import List
 
+import torch
+import torch.nn.functional as F
+from sentence_transformers import SentenceTransformer
+from torch import Tensor
+from tqdm import tqdm
+
+from torch_frame import stype
+from torch_frame.config.text_embedder import TextEmbedderConfig
+from torch_frame.data import DataLoader
 from torch_frame.datasets import Mercari
+from torch_frame.nn import (
+    EmbeddingEncoder,
+    FTTransformer,
+    LinearEmbeddingEncoder,
+    LinearEncoder,
+)
+from torch_frame.nn.encoder.stype_encoder import \
+    MultiCategoricalEmbeddingEncoder  # noqa
+
+
+class PretrainedTextEncoder:
+    def __init__(self, device: torch.device):
+        self.model = SentenceTransformer('all-distilroberta-v1', device=device)
+
+    def __call__(self, sentences: List[str]) -> Tensor:
+        # Inference on GPU (if available)
+        embeddings = self.model.encode(sentences, convert_to_numpy=False,
+                                       convert_to_tensor=True)
+        # Map back to CPU
+        return embeddings.cpu()
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--channels', type=int, default=256)
+parser.add_argument('--num_layers', type=int, default=4)
+parser.add_argument('--batch_size', type=int, default=512)
+parser.add_argument('--lr', type=float, default=0.0001)
+parser.add_argument('--epochs', type=int, default=100)
+parser.add_argument('--seed', type=int, default=0)
+args = parser.parse_args()
+
+torch.manual_seed(args.seed)
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+
+# Prepare datasets
+path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'mercari')
+text_encoder = PretrainedTextEncoder(device=device)
+dataset = Mercari(
+    root=path,
+    num_rows=5000,  # delete the argument to use the full dataset
```

Comment:
Can we remove this and report performance against public leaderboard before merging?